revolution_pi/gpio_pin.go

```diff
@@ -105,17 +187,27 @@ func (pin *gpioPin) PWM(ctx context.Context, extra map[string]interface{}) (floa
 	b[1] = 0x00
 	val := binary.LittleEndian.Uint16(b)
 	if val > 100 {
-		pin.ControlChip.logger.Info("got PWM duty cycle greater than 100")
+		pin.ControlChip.logger.Warn("got PWM duty cycle greater than 100")
 	}
-	return float64(val), nil
+	return float64(val) / 100, nil
 }
 
 // SetPWM sets the pin to the given duty cycle.
 func (pin *gpioPin) SetPWM(ctx context.Context, dutyCyclePct float64, extra map[string]interface{}) error {
 	if !pin.initialized {
 		return errors.New("pin not initialized")
 	}
+	if !pin.isOutputPWM() && !pin.isOutputWord() {
+		return fmt.Errorf("cannot set PWM, Pin %s is not a PWM pin", pin.Name)
+	}
+
+	// if the pin isn't in PWM mode, throw an error
+	if !pin.pwmMode {
+		return fmt.Errorf("cannot set PWM, Pin %s is not configured for PWM", pin.Name)
```

Comment:
This code has been copied and pasted from above. I wonder if it would be useful to put it in a helper function.
```go
func (pin *gpioPin) canSupportPWM() error {
    ...
}
```

Reply:
tried adding a helper but I couldn't figure out how I would want to use it